dotnet/src/SemanticKernel/Planning/IPlan.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
```

Comment:
I think this can be simplified, avoiding also the need for a new interface. In fact, I would argue that a plan should implement ISKFunction.  Here's a proposal, keeping it simple and showing that a Plan is a Function.

```
public class Plan : ISKFunction
{
    public ContextVariables State { get; } = new();

    public List<Plan> Steps { get; } = new();

    public Task<IPlan> RunNextStepAsync(IKernel kernel, ContextVariables variables, CancellationToken cancellationToken = default) { }

    // == ISKFunction interface ==

    public string Name { get; } // == SelectedFunction

    public string SkillName { get; } // == SelectedSkill

    public string Description { get; } // == Goal description

    public bool IsSemantic { get; } = false;

    public CompleteRequestSettings RequestSettings { get; } // unused

    // this allows to use the plan as a function, see previous PR comment
    public FunctionView Describe() { }

    // this allows to execute the entire plan
    public Task<SKContext> InvokeAsync(string input, SKContext? context = null, CompleteRequestSettings? settings = null, ILogger? log = null, CancellationToken? cancel = null) { }

    // this allows to execute the entire plan
    public Task<SKContext> InvokeAsync(SKContext? context = null, CompleteRequestSettings? settings = null, ILogger? log = null, CancellationToken? cancel = null) { }

    // used by the kernel to map names
    public ISKFunction SetDefaultSkillCollection(IReadOnlySkillCollection skills) { }

    // reserved to semantic functions
    public ISKFunction SetAIService(Func<ITextCompletion> serviceFactory) { }

    // reserved to semantic functions
    public ISKFunction SetAIConfiguration(CompleteRequestSettings settings) { }
}
```

Reply:
Did this.